### PR TITLE
feat: add injection framework, inject favicon and duck

### PR DESCRIPTION
**Pull request by MiniDigger**

### CODE REVIEW

**Comment by powercasgamer** [comment]
For the favicon injection it should probably check if the favicon exists before trying to add it. Perhaps something like [this?](https://github.com/seiama/javaducks/commit/d1e59b1a49df7a3f2946e3e799af52bdffc44fa2)

**Comment by MiniDigger** [comment]
> For the favicon injection it should probably check if the favicon exists before trying to add it. Perhaps something like [this?](https://github.com/seiama/javaducks/commit/d1e59b1a49df7a3f2946e3e799af52bdffc44fa2)

I dont think that makes a difference (the browser will always try to fetch a favicon, even if you dont define one), I did fix the bug that caused a 500 to be thrown when trying to request a non existing favicon tho.

File: src/main/java/com/seiama/javaducks/injection/OutdatedBannerInjection.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+@Component
+public class OutdatedBannerInjection implements Injection {
+
+  private final AppConfiguration configuration;
+  private final String template;
+
+  public OutdatedBannerInjection(final AppConfiguration configuration) {
+    this.configuration = configuration;
+    try {
+      this.template = Files.readString(Path.of(Objects.requireNonNull(getClass().getClassLoader().getResource("outdated-banner.html")).toURI()));
```
Comment (kashike) [inline src/main/java/com/seiama/javaducks/injection/OutdatedBannerInjection.java]: Can we use a [`Resource`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/Resource.html)/[`ClassPathResource`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/ClassPathResource.html) instead?

File: src/main/java/com/seiama/javaducks/service/InjectionService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+public class InjectionService {
```
Comment (kashike) [inline src/main/java/com/seiama/javaducks/service/InjectionService.java]:
```suggestion
public class JavadocInjector {
```

File: src/main/java/com/seiama/javaducks/controller/JavadocController.java [hunk 3]
```diff
@@ -65,10 +69,12 @@ public class JavadocController {
     ".html", MediaType.parseMediaType("text/html")
   );
   private final JavadocService service;
+  private final InjectionService injectionService;
```
Comment (kashike) [inline src/main/java/com/seiama/javaducks/controller/JavadocController.java]:
```suggestion
  private final JavadocInjector injector;
```

**Review by kashike — CHANGES_REQUESTED** [review]

File: src/main/java/com/seiama/javaducks/injection/FaviconInjection.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.seiama.javaducks.service.JavadocService;
+import java.nio.file.Path;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FaviconInjection implements Injection {
+
+  @Override
+  public boolean canInject(final Path file, final JavadocService.Key key) {
+    return file.toString().endsWith(".html");
```
Comment (kashike) [inline src/main/java/com/seiama/javaducks/injection/FaviconInjection.java]: Constant for `".html"`.

File: src/main/java/com/seiama/javaducks/injection/DuckInjection.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.seiama.javaducks.service.JavadocService;
+import java.nio.file.Path;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DuckInjection implements Injection {
+  @Override
+  public boolean canInject(final Path file, final JavadocService.Key key) {
+    return file.toString().endsWith(".html");
```
Comment (powercasgamer) [inline src/main/java/com/seiama/javaducks/injection/DuckInjection.java]:
```suggestion
    return file.toString().endsWith(HTML);
```

**Review by powercasgamer — CHANGES_REQUESTED** [review]

File: src/main/java/com/seiama/javaducks/injection/DuckInjection.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ */
+package com.seiama.javaducks.injection;
+
+import com.seiama.javaducks.service.JavadocService;
+import java.nio.file.Path;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DuckInjection implements Injection {
+  @Override
```
Comment (powercasgamer) [inline src/main/java/com/seiama/javaducks/injection/DuckInjection.java]:
```suggestion
public class DuckInjection implements Injection {

  @Override
```

**Review by powercasgamer — CHANGES_REQUESTED** [review]
To match the other injectors

File: src/main/java/com/seiama/javaducks/injection/DuckInjection.java (hunk 5 above)
Comment (MiniDigger) [reply src/main/java/com/seiama/javaducks/injection/DuckInjection.java]: good catch

**Review by kashike — APPROVED** [review]

**Pull request merged**